cmd/thanos/compact.go

```diff
@@ -136,8 +141,10 @@ func runCompact(
 
 		ctx, cancel := context.WithCancel(context.Background())
 
-		g.Add(func() error {
-			return runutil.Repeat(5*time.Minute, ctx.Done(), func() error {
+		f := func() error {
+			// Loop over bucket and compact until there's no work left.
+			didWork := false
+			for {
```

Comment:
just do `for didWork {` and kill 
```
if !didWork { ...
```
part?

Reply:
We want `do ... while` semantics though and not `while`.

Reply:
yea so change variable name to `workLeft` and put `for workLeft {` (: